### DISCUSSION TITLE
A lot of "Connection reset by peer for" messages in the log

### DISCUSSION THREAD

**Discussion by sellmann** [s-allius/tsun-gen3-proxy#89] [Q&A]
Hello,

first of all thank you very much for the proxy. Good work!

I'm using the proxy in a docker container on my raspberry pi 4 (arm64 image) together with mosquitto mqtt broker and OpenHab also running in containers. I have added the DNS entry  for logger.talent... in my pi-hole and a dns rebind entry into my Fritzbox.

My problem is, that the mqtt transmission (e.g. tsun-topics) stops after a while. In the proxy log there are a lot of connection reset messages and "Host is not reachable". 172.18.0.2 is the IP address of the proxy container (bridge network), 47.254.132.252 seems to be the address of logger.talent:

'2024-06-12 08:55:36 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.18.0.2', 55086) | r('47.254.132.252', 5005)'
'2024-06-12 08:55:36  INFO | root | [] Client loop stopped for l('172.18.0.2', 55086)'
'2024-06-12 08:57:08  INFO | root | [] Connected to ('logger.talent-monitoring.com', 5005)'
'2024-06-12 09:00:58 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.18.0.2', 39454) | r('47.254.132.252', 5005)'
'2024-06-12 09:00:58  INFO | root | [] Client loop stopped for l('172.18.0.2', 39454)'
'2024-06-12 09:02:26  INFO | root | [] Connected to ('logger.talent-monitoring.com', 5005)'
'2024-06-12 09:06:20 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.18.0.2', 59334) | r('47.254.132.252', 5005)'
'2024-06-12 09:06:20  INFO | root | [] Client loop stopped for l('172.18.0.2', 59334)'
'2024-06-12 09:07:40  INFO | root | [] Connected to ('logger.talent-monitoring.com', 5005)'
'2024-06-12 09:11:06 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.18.0.2', 49476) | r('47.254.132.252', 5005)'
'2024-06-12 09:11:06  INFO | root | [] Client loop stopped for l('172.18.0.2', 49476)'
'2024-06-12 09:12:57  INFO | root | [] Connected to ('logger.talent-monitoring.com', 5005)'
'2024-06-12 09:16:59 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.18.0.2', 37812) | r('47.254.132.252', 5005)'
'2024-06-12 09:16:59  INFO | root | [] Client loop stopped for l('172.18.0.2', 37812)'
'2024-06-12 09:37:36 ERROR | conn | [] [Errno 113] Host is unreachable for l('172.18.0.2', 5005) | r('192.168.178.101', 53274)'
'2024-06-12 09:37:36  INFO | root | [] Server loop stopped for r('192.168.178.101', 53274)'

What can I do to resolve the issue?

Thank you very much.

**Comment by s-allius** [comment]
Hi Sellmann,

I think you can't do or fix this situation. It's seems for me that the TSUN close has a problem.

The proxy has two connection. One is called the "server side", which is the connection to the inverter, and the "client side" which is the connection to the TSUN cloud. The message "connection reset by peer" means that the remote side has disconnected, the connection and in the next line you can read, that the client loop (connection) was stopped. So the TSUN cloud send as an `RST` on the TCP connection, which we can#t solve. With the next packet for the TSUN cloud we will establish a new connection.

So the only solution ist, don't connect to TSUN and run everything on your own.

**Comment by sellmann** [comment]
Thank you for your response. I will disable the cloud connection because I don't need it. Instead of Talent Home or Pro App I will use my smart home solution. As you can see the Server Loop also stopped working and the Inverter went offline. Will this happen again without the cloud connection?

**Reply by s-allius** [reply]
Loosing the server connection happens sometime (rarely), especially on GEN3 inverters. In the trace above it seems to me that is a follow up error of 'server not reachable ... 192.168.178.101, 53274'. In this situation the proxy disconnect the connection. That will not happen if you disable connection to the cloud.
Pls give a feedback, if your problems are solved by disabling the cloud connection.

**Reply by s-allius** [reply]
I thought a little bit about your traces and found a problem, when the connection to the TSUN cloud get lost, the connection to the inverter will be disconnected by the proxy in some situations. I will open an issue for that

**Reply by sellmann** [reply]
Thank you. Today the inverter is offline most of the time and does not reconnect. I don't think the cause is the proxy but I paste the log in here just in case. I restarted the proxy at 09:23 am and it took until 14:31 pm when the inverter connected. At 15:41 pm the server loop stopped again because the inverter was not rachable any longer.

'2024-06-17 09:23:38  INFO | root | Server "tsun-proxy - 0.8.0" will be started'
'2024-06-17 09:23:38  INFO | mqtt | start MQTT: host:192.168.178.122  port:1883  user:None'
'2024-06-17 09:23:38  INFO | mqtt | MQTT broker connection established'
'2024-06-17 09:23:38  INFO | root | Initialize proxy device on home assistant'
'2024-06-17 14:31:57  INFO | root | [G3] Accept connection from ('192.168.178.101', 63122)'
'2024-06-17 15:41:20 ERROR | conn | [] [Errno 113] Host is unreachable for l('172.17.0.6', 5005) | r('192.168.178.101', 63122)'
'2024-06-17 15:41:20  INFO | root | [] Server loop stopped for r('192.168.178.101', 63122)'

**Reply by s-allius** [reply]
How good is the connection quality to your inverter? Host unreachable is a hint that you have a connection problem, because the TCP connection was lost due to a routing problem, or an other common problem.

**Reply by sellmann** [reply]
The signal strength is 84% according to the TSUN website and I have a wlan repater in my garden with an individual wlan set where the inverter connects to. I don't know what else to do, the G3 inverter has some flaws including the regular connection drop once a day. I'm a silent reader of the corresonding thread in the photovoltaik forum...

**Comment by sellmann** [comment]
Without the TSUN cloud the connection to the inverter is much more stable. The inverter seems to close the connection in regular intervals (aprox. once in a hour) and connects again like in the log sample below:

'2024-06-15 17:12:55  INFO | root | [G3] Accept connection from ('192.168.178.101', 64522)'
'2024-06-15 18:00:06 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.17.0.6', 5005) | r('192.168.178.101', 64522)'
'2024-06-15 18:00:06  INFO | root | [] Server loop stopped for r('192.168.178.101', 64522)'
'2024-06-15 18:00:06  INFO | root | [G3] Accept connection from ('192.168.178.101', 55748)'
'2024-06-15 18:52:20 ERROR | conn | [] [Errno 104] Connection reset by peer for l('172.17.0.6', 5005) | r('192.168.178.101', 55748)'
'2024-06-15 18:52:20  INFO | root | [] Server loop stopped for r('192.168.178.101', 55748)'
'2024-06-15 18:52:21  INFO | root | [G3] Accept connection from ('192.168.178.101', 65170)'

You don't notice anything and the mqtt topics are still refreshed every minute.

**Reply by s-allius** [reply]
with the next version 0.9.0 comes a unique connection id in the logs. This will help to understand which connection logs which message and which Is the root cause of an action

**Comment by s-allius** [comment]
Maybe I found the problem.
The GEN3 inverter seems to have problems when we send a second MODBUS command directly after we received a MODBUS response. During the tests of the version 0.8.0 I never saw this behaviour. I have prepared a hotfix for that. When the test tomorrow are passes I will release a fix version 0.8.1 soon

**Reply by sellmann** [reply]
This sounds good. I will update the image as soon as the hotfix is available. I appreciate your help and work. Thank you!

**Comment by sellmann** [comment]
I suddenly discovered another strange behavior. I noticed some cpu work on my raspberry pi and the proxy seems to loop, take a look at the log:

'2024-06-19 08:40:30  INFO | root | Initialize proxy device on home assistant'
'2024-06-19 08:40:30  INFO | mqtt | MQTT broker connection established'
'2024-06-19 08:40:30  INFO | root | Initialize proxy device on home assistant'
'2024-06-19 08:40:31  INFO | mqtt | MQTT broker connection established'
'2024-06-19 08:40:31  INFO | root | Initialize proxy device on home assistant'
'2024-06-19 08:40:31  INFO | mqtt | MQTT broker connection established'
'2024-06-19 08:40:31  INFO | root | Initialize proxy device on home assistant'
...

The log is full of these entries. I had to stop and restart the container.

**Reply by s-allius** [reply]
After restarting the container the problem was gone?

**Reply by sellmann** [reply]
Yes.

**Reply by sellmann** [reply]
I also found this type of log entries:
2024-06-19 20:31:27 ERROR | root | Infos.parse: data_type: 65 not supported'

**Reply by s-allius** [reply]
It would be very nice if you can send me the trace.log of this packet. I only need about 3 minutes before and 3 minutes after this packet.
Pls send the trace to 'unknown_msg@allius.de', than I can look how to support this.

**Reply by sellmann** [reply]
Done

**Reply by s-allius** [reply]
Thank you. I will bring a fix with the next release (0.9.0). You have a newer DSP firmware with version V5.0.17, which brings some new data encodings.
